server/src/main/java/org/elasticsearch/ingest/IngestService.java

```diff
@@ -307,16 +290,9 @@ static void resolvePipelinesAndUpdateIndexRequest(
          * we don't fall back to the existing index if we don't find any because it is possible the user has intentionally removed the
          * pipeline.
          */
-        final Pipelines pipelines;
```

Comment:
We need to update this comment and mention that in order of the simulation to work we would need to remove any existing index/data stream. I am thinking that now this can easily be missed.

Reply:
Oh good point! I had forgotten about this comment.